src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/validation/ItemGroupsValidator.java

```diff
@@ -1,97 +1,142 @@
         }
 
-        return result;
+        return true;
     }
 
-    private void validateCompanyNumber(ItemGroupData dto, List<String> errors) {
-    }
+    private boolean linksPresentWithOrderNumber(ItemGroupData dto, List<String> errors) {
+        Links links = dto.getLinks();
 
-    private void validateItems(ItemGroupData dto, List<String> errors) {
-        var items = dto.getItems();
+        if (isNull(links)) {
+            errors.add(ITEM_GROUP_LINKS_MISSING);
+            return false;
+        }
 
-        if (items == null || items.isEmpty()) {
-            errors.add("items missing");
+        if (StringUtils.isBlank(links.getOrder())) {
+            errors.add(LINKS_ORDER_NUMBER_MISSING);
+            return false;
         }
-    }
 
-    private void validateLinks(ItemGroupData dto, List<String> errors) {
-        Links links = dto.getLinks();
+        return true;
+    }
+    /**
+     * Taken from: <a href="https://developer-specs.cidev.aws.chdev.org/item-group-workflow-api/reference/itemgroups/create-item-group">create item group</a>
+     * Optional : The company name associated with this item (mandatory field if the item relates to a specific company)
+     * Optional : The company number associated with this item (mandatory field if the item relates to a specific company)
+     * Both flagged as optional and 'mandatory field if the item relates to a specific company'
+     * Working assumption : if delivery_details.company_name is present then both these fields are mandatory.
+     */
+    private void validateDeliveryDetailsCompanyNumberAndName(ItemGroupData dto, List<String> errors) {
+        DeliveryDetails deliveryDetails = dto.getDeliveryDetails();
+        if (isNull(deliveryDetails))
+            return;
 
-        if (links == null) {
-            errors.add("links missing");
+        String deliveryCompanyName = deliveryDetails.getCompanyName();
+        if (StringUtils.isBlank(deliveryCompanyName))
             return;
-        }
 
-        String order = links.getOrder();
+        for (Item item : dto.getItems()) {
+            if (StringUtils.isBlank(item.getCompanyNumber())) {
+                errors.add(COMPANY_NUMBER_MISSING + deliveryCompanyName);
+            }
 
-        if (order == null || order.isBlank() || order.isEmpty()) {
-            errors.add("links missing order number");
+            if (StringUtils.isBlank(item.getCompanyName())) {
+                errors.add(COMPANY_NAME_MISSING + deliveryCompanyName);
+            }
         }
     }
 
-    private void validateDescriptionIdentifier(ItemGroupData dto, List<String> errors) {
-        // TODO - DCAC-47 - for future validation.
+    private void validateItemDescriptionIdentifier(ItemGroupData dto, List<String> errors) {
+        for (Item item : dto.getItems()) {
+            String descriptionIdentifier = item.getDescriptionIdentifier();
+
+            if (isNull(ItemDescriptionIdentifier.getEnumValue(descriptionIdentifier))) {
```

Comment:
Did you look into using an annotation like `@ValueOfEnum(enumClass = ItemDescriptionIdentifier.class)`?